src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -49,12 +52,44 @@ public IslandManager() {
         super(IridiumSkyblock.getInstance());
     }
 
-    public void createWorld(World.Environment environment, String name) {
+    public void createWorld(World.Environment environment, String name) throws IOException {
         if (!IridiumSkyblock.getInstance().getConfiguration().enabledWorlds.getOrDefault(environment, true)) return;
         WorldCreator worldCreator = new WorldCreator(name)
                 .generator(IridiumSkyblock.getInstance().getDefaultWorldGenerator(name, null))
                 .environment(environment);
         Bukkit.createWorld(worldCreator);
+
+        if(Bukkit.getWorld(worldCreator.name()).getEnvironment() == World.Environment.THE_END) {
+
+            Bukkit.unloadWorld(worldCreator.name(), true);
+
+            File file = new File(worldCreator.name() + File.separator + "level.dat");
+            NBTFile worldFile = new NBTFile(file);
+
+            if(worldFile.getCompound("Data").getCompound("DragonFight") == null) {
+                IridiumSkyblock.getInstance().getLogger().warning("Cannot load \"DragonFight\" compound because \"DragonFight\" is null.");
+                return;
+            }
+
+            NBTCompound compound = worldFile.getCompound("Data").getCompound("DragonFight");
+            Byte PreviouslyKilled = compound.getByte("PreviouslyKilled");
+            Byte DragonKilled = compound.getByte("DragonKilled");
+            Byte NeedsStateScanning = compound.getByte("NeedsStateScanning");
+
+            if(PreviouslyKilled == (byte) 0) {
+                compound.setByte("PreviouslyKilled", (byte) 1);
+            }
+            if(DragonKilled == 0) {
+                compound.setByte("DragonKilled", (byte) 1);
+            }
+            if(NeedsStateScanning == 1) {
+                compound.setByte("NeedsStateScanning", (byte) 0);
+            }
```

Comment:
Can we not change this to setBoolean? iirc it just wraps setByte and sets it to 0/1

Reply:
Also, you can probabbly remove the if statements no?

Reply:
i haven't tried that actually, i'll give it a shot and see what we end up with (this was done based on me looking at the NBTdata directly so)